File: app/src/main/java/dev/ragnarok/fenrir/api/services/IStatusService.java
package dev.ragnarok.fenrir.api.services;

import dev.ragnarok.fenrir.api.model.response.BaseResponse;
import io.reactivex.rxjava3.core.Single;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;


public interface IStatusService {

    /**
     * Sets a new status for the current user.
     *
     * @param text    Text of the new status.
     * @param groupId Identifier of a community to set a status in. If left blank the status is set to current user.
     * @return 1
     */
    @FormUrlEncoded
    @POST("status.set")
    Single<BaseResponse<Integer>> set(@Field("text") String text,
                                      @Field("group_id") Integer groupId);
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/api/services/IStatusService.kt
package dev.ragnarok.fenrir.api.services

import dev.ragnarok.fenrir.api.model.response.BaseResponse
import dev.ragnarok.fenrir.api.rest.IServiceRest
import io.reactivex.rxjava3.core.Single

class IStatusService : IServiceRest() {
    /**
     * Sets a new status for the current user.
     *
     * @param text    Text of the new status.
     * @param groupId Identifier of a community to set a status in. If left blank the status is set to current user.
     * @return 1
     */
    operator fun set(
        text: String?,
        groupId: Long?
    ): Single<BaseResponse<Int>> {
        return rest.request(
            "status.set", form(
                "text" to text,
                "group_id" to groupId
            ), baseInt
        )
    }
}